File: Insertion/insertion.py
def sort(lst):
    """
    Sorts lst in place with no return using Insertion Sort
    """
    for i in range(len(lst)):
        pos = i
        for j in range(i)[::-1]:
            if lst[pos] < lst[j]:
                swap(pos, j, lst)
                pos = j
            else:
                break
def swap(i, j, lst):
    temp = lst[i]
    lst[i] = lst[j]
    lst[j] = temp

File: Insertion/insertion.go
package main

import "fmt"

func main() {
	var test = []int{3,2,1, 4}
	fmt.Printf("%v\n", test)
	sort(test)
	fmt.Printf("%v\n", test)
}

func sort(xs []int){
	for k,_ := range xs {
		a := k
		for j := k-1; j>= 0; j-- {
			if xs[a] < xs[j] {
				temp := xs[a]
				xs[a] = xs[j]
				xs[j] = temp
				a = j
			}
		}
	}
}
